### PR TITLE
Persistent Menu Navigation in Web Apps

### PR DESCRIPTION
**Pull request by esoergel**
## Product Description

Adds a pop-out drawer containing a persistent menu.

Collapsed view:
image

Expanded view:
image

## Technical Summary
[Jira Ticket](https://dimagi.atlassian.net/browse/USH-4711)
[Product Spec](https://docs.google.com/document/d/1MMsRk2aXXTsqPtL5e2Sl9EJdTYGi5mrYfKIls4ct1LU/edit)
[Tech Spec](https://docs.google.com/document/d/12Y4CO-Sl-yTUsdO5VfvDgluRmSXjRTl7FTosQNhyP7M/edit#heading=h.8nqx0ufwj60c)
Branched of of https://github.com/dimagi/commcare-hq/pull/34682

I'd review by commit - commit messages have context for each change.

## Feature Flag
PERSISTENT_MENU_SETTING: Show Persistent Menu option in Web Apps settings

### Safety story

This will be going through QA.  It's a highly visible change, though.

### QA Plan

https://dimagi.atlassian.net/browse/QA-6819

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/cloudcare/templates/cloudcare/formplayer_home.html [hunk 1]
```diff
@@ -98,22 +98,24 @@
     <div id="menu-container">
       <section id="formplayer-progress-container"></section>
       <div id="restore-as-region"></div>
-      <div id="breadcrumb-region" class="print-container"></div>
-      <section id="cloudcare-notifications" class="container notifications-container"></section>
-      <div class="container case-tile-container">
-        <div id="persistent-case-tile" class="print-container"></div>
+      <div id="cloudcare-app-context">
```
Comment (esoergel) [inline corehq/apps/cloudcare/templates/cloudcare/formplayer_home.html:101]: Hide whitespace on this commit - I just wrapped an existing block of code

File: corehq/apps/hqwebapp/static/cloudcare/scss/formplayer-webapp/menu.scss [hunk 2]
```diff
@@ -12,3 +12,15 @@
...
+  background-color: rgb(0 0 0 / 25%);
+  position: fixed;
+  top: 0;
+  left: 0;
+  z-index: 1000;
+
+  #persistent-menu-content {
+    background-color: beige;
+  }
+}
```
Comment (esoergel) [inline corehq/apps/hqwebapp/static/cloudcare/scss/formplayer-webapp/menu.scss]: This was all temporary stuff in helping me build this out - it gets replaced later

**Review by AddisonDunn — DISMISSED** [review]

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js [hunk 3]
```diff
@@ -161,11 +161,29 @@ hqDefine("cloudcare/js/formplayer/menus/controller", [
         } else {
             FormplayerFrontend.regions.getRegion('breadcrumb').empty();
         }
+
+        if (menuResponse.persistentMenu) {
```
[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js:165]: not at present, but that's the eventual goal.

File: corehq/apps/cloudcare/templates/cloudcare/partials/menu/persistent_menu.html [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+<script type="text/template" id="persistent-menu-template">
+  <div type="button"
+          data-bs-toggle="offcanvas"
+          data-bs-target="#persistent-menu-container"
+          aria-expanded="false"
+          aria-controls="persistent-menu-container"
+          class="position-fixed top-50 start-0 rounded-end persistent-menu-toggle">
+    <i class="fa fa-chevron-right"></i>
+  </div>
+  <div id="persistent-menu-container" class="offcanvas offcanvas-start" tabindex="-1" aria-labelledby="persistent-menu-label">
```
[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/apps/cloudcare/templates/cloudcare/partials/menu/persistent_menu.html:12]: I didn't actually test it with a screen reader, but it's supposed to be.  Good to confirm in QA.

File: corehq/apps/cloudcare/templates/cloudcare/partials/menu/persistent_menu.html [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+<script type="text/template" id="persistent-menu-item">
+  <div>
+    <% if (imageUri) { %>
+      <i class="fcc persistent-menu-image" aria-hidden="true"
+         style="background-image: url('<%- imageUri %>');"></i>
+    <% } else { %>
+      <i class="fa fa-arrow-up-right-from-square persistent-menu-image" aria-hidden="true"></i>
+    <% } %>
+    <a class="align-middle" href="#"><%- displayText %></a>
```
[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/apps/cloudcare/templates/cloudcare/partials/menu/persistent_menu.html:39]: I've been browsing the B5 docs and learning a lot!

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js [hunk 6]
```diff
@@ -1489,6 +1495,88 @@ hqDefine("cloudcare/js/formplayer/menus/views", [
...
+        triggers: {
+            "click a": "click:persistent:menu:command"  // magically calls onClickPersistentMenuCommand
+        },
+        templateContext: function () {
+            const appId = formplayerUtils.currentUrlToObject().appId,
+                  imageUri = this.model.get('imageUri'),
+                  icons = {JUMP: 'fa-pencil', NEXT: 'fa-regular fa-folder'};
+            return {
+                imageUri: imageUri ? FormplayerFrontend.getChannel().request('resourceMap', imageUri, appId) : "",
+                iconClass: icons[this.model.get('navigationState')] || 'fa-list-ul',
```
Comment (Jtang-1) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js:1517]: FYI we decided to explicitly make a navigationState for case list. It'll be `ENTITY_SELECT` https://github.com/dimagi/formplayer/pull/1603/commits/0f2f63947b0afb26976fe7fd800b7236103bd10f

File: corehq/apps/cloudcare/templates/cloudcare/partials/menu/persistent_menu.html (hunk 4 above)
[the remark this answers is not in the document]
Comment (esoergel) [reply corehq/apps/cloudcare/templates/cloudcare/partials/menu/persistent_menu.html:14]: Did a bunch of fiddling to get the tab navigation stuff right, but it should be good now.

File: corehq/apps/app_manager/templates/app_manager/profile.xml [hunk 7]
```diff
@@ -29,8 +29,8 @@
     <property key="support-email-address" value="{{ support_email }}" force="true"/>
     {% endif %}
 
-    <property key="cc_persistent_menu" value="{{ app.persistent_menu|yesno:"yes,no" }}" force="true"/>
-    <property key="cc_breadcrumbs_enabled" value="{{ app.show_breadcrumbs|yesno:"yes,no" }}" force="true"/>
+    <property key="cc-persistent-menu" value="{{ app.persistent_menu|yesno:"yes,no" }}" force="true"/>
```
Comment (MartinRiese) [inline corehq/apps/app_manager/templates/app_manager/profile.xml:32]: Is FP expecting this change?

**Review by MartinRiese — APPROVED** [review]

File: corehq/apps/app_manager/templates/app_manager/profile.xml (hunk 7 above)
Comment (esoergel) [reply corehq/apps/app_manager/templates/app_manager/profile.xml:32]: Yep, I made it at Shubham's request

**Review by MartinRiese — APPROVED** [review]

**Pull request merged**